node/src/components/block_synchronizer/block_builder/tests.rs

```diff
@@ -0,0 +1,198 @@
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder.handle_acceptance(None, Ok(None)).is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(None))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(Some(Acceptance::HadIt)))
+        .is_ok());
+    assert!(builder.peer_list().qualified_peers(&mut rng).is_empty());
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(Some(honest_peer), Ok(Some(Acceptance::NeededIt)))
+        .is_ok());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(None, Err(BlockAcquisitionError::InvalidStateTransition))
+        .is_err());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder.peer_list().dishonest_peers().is_empty());
+
+    assert!(builder
+        .handle_acceptance(
+            Some(dishonest_peer),
+            Err(BlockAcquisitionError::InvalidStateTransition)
+        )
+        .is_err());
+    assert!(builder
+        .peer_list()
+        .qualified_peers(&mut rng)
+        .contains(&honest_peer));
+    assert!(builder
+        .peer_list()
+        .dishonest_peers()
+        .contains(&dishonest_peer));
+}
+
+#[test]
+fn register_era_validator_weights() {
+    let mut rng = TestRng::new();
+    let block = Block::random(&mut rng);
+    let mut builder = BlockBuilder::new(
+        block.header().block_hash(),
+        false,
+        1,
+        TimeDiff::from_seconds(1),
+        TimeDiff::from_seconds(1),
+    );
+
+    let mut validator_matrix = ValidatorMatrix::new_with_validator(ALICE_SECRET_KEY.clone());
+
+    builder.register_era_validator_weights(&validator_matrix);
+    assert!(builder.validator_weights.is_none());
+
+    builder.era_id = Some(EraId::from(1000));
+    builder.register_era_validator_weights(&validator_matrix);
+    assert!(builder.validator_weights.is_none());
+
+    builder.era_id = Some(block.header().era_id());
+    let weights = EraValidatorWeights::new(
+        block.header().era_id(),
+        BTreeMap::from([(ALICE_PUBLIC_KEY.clone(), 100.into())]),
+        Ratio::new(1, 3),
+    );
+    validator_matrix.register_era_validator_weights(weights.clone());
+    builder.register_era_validator_weights(&validator_matrix);
+    assert_eq!(builder.validator_weights.unwrap(), weights);
+}
+
+#[test]
+fn register_block_execution_enqueued() {
```

Comment:
Please consider covering the scenario when a call to `BlockAcquititionState::register_block_execution_enqueued()` returns an `Err`.

Reply:
Done.